### PR TITLE
[examples] fix summarization do_predict

### LINKED ISSUE #3853: How to use fine-tuned BART for prediction?
# ❓ Questions & Help

## Details
I fine-tuned the BART model on a custom summarization dataset using the **transformers/examples/summarization/bart/finetune.py** and **transformers/examples/summarization/bart/run_train.sh** files in the repository for training (which generated three _checkpointepoch=*.ckpt_ files) and prediction (which generated a _.txt_ file with the test loss scores). 

I have two questions on using this model for prediction:

- How can I modify _finetune.py_ to generate predictions for the test set, in addition to the loss scores? I see some test functions in _finetune.py_, but I'm not sure how to use these for generating a _.txt_ file with the predictions.

- How can I load the generated _.ckpt_ files into BartForConditionalGeneration()? A _config.json_ file was not generated along with the checkpoint files; there doesn't seem to be a TFBartForConditionalGeneration; and the _convert_tf_checkpoint_to_pytorch.py_ script in the repo doesn't seem to support BART yet.

Thank you for your time!

### LINKED ISSUE #3865: Summarisation tuning
Hi everybody,

I’ve tried using BART summarisation code, and I had a question about finetune.py 

Can SummarisationTrainer checkpoint be loaded as a BartForConditionalGeneration model from the evaluation script?

### PR DESCRIPTION
**Pull request by sshleifer**
- by copying NER
- add type hints

### CODE REVIEW

File: examples/summarization/bart/finetune.py [hunk 1]
```diff
@@ -166,8 +166,12 @@ def main(args):
 
     # Optionally, predict on dev set and write to output_dir
     if args.do_predict:
+        # See https://github.com/huggingface/transformers/issues/3159
+        # pl use this format to create a checkpoint:
+        # https://github.com/PyTorchLightning/pytorch-lightning/blob/master\
+        # /pytorch_lightning/callbacks/model_checkpoint.py#L169
         checkpoints = list(sorted(glob.glob(os.path.join(args.output_dir, "checkpointepoch=*.ckpt"), recursive=True)))
-        SummarizationTrainer.load_from_checkpoint(checkpoints[-1])
+        model = model.load_from_checkpoint(checkpoints[-1])
```
Comment (sshleifer) [inline examples/summarization/bart/finetune.py:174]: This is the fix.

**Review by thomwolf — APPROVED** [review]
